web/pandas/pdeps/0005-no-default-index-mode.md

```diff
@@ -0,0 +1,189 @@
+# PDEP-5: No-default-index mode
+
+- Created: 14 November 2022
+- Status: Draft
+- Discussion: [#49693](https://github.com/pandas-dev/pandas/pull/49693)
+- Author: [Marco Gorelli](https://github.com/MarcoGorelli)
+- Revision: 1
+
+## Abstract
+
+The suggestion is to add a `mode.no_default_index` option which, if enabled,
+would ensure:
+- if a ``DataFrame`` / ``Series`` is created, then by default it won't have an ``Index``;
+- nobody will get an ``Index`` unless they ask for one - this would affect the default behaviour of ``groupby``, ``value_counts``, ``pivot_table``, and more.
+
+This option would not be the default. Users would need to explicitly opt-in to it, via ``pd.set_option('mode.no_default_index', True)``, via ``pd.option_context``, or via the ``PANDAS_NO_DEFAULT_INDEX`` environment variable.
+
+## Motivation and Scope
+
+The Index can be a source of confusion and frustration for pandas users. For example, let's consider the inputs
+
+```python
+In [37]: ser1 = df.groupby('sender')['amount'].sum()
+
+In [38]: ser2 = df.groupby('receiver')['amount'].sum()
+
+In [39]: ser1
+Out[39]:
+sender
+1    10
+2    15
+3    20
+5    25
+Name: amount, dtype: int64
+
+In [40]: ser2
+Out[40]:
+receiver
+1    10
+2    15
+3    20
+4    25
+Name: amount, dtype: int64
+```
+. Then:
+
+- it can be unexpected that summing `Series` with the same length (but different indices) produces `NaN`s in the result (https://stackoverflow.com/q/66094702/4451315):
```

Comment:
I always thought this was specifically by design and if one wanted strictly location-based addition, they should just use NumPy.

Reply:
That may have been a thought 10+ years ago, but that's not realistic today, and never has been. Many users use pandas as something like "NumPy for 2-D data + labeled axes + sane strings + groupby and other methods that make sense for tabular data". This ship sailed a long time ago, when NumPy decided to not implement `groupby` and labelled axes, because folks were happy using pandas for those.

Reply:
It is by design, but it's still unintuitive

In a (totally unscientific) poll I asked a group of experienced developers what the output of
```python
ser1 = pd.Series([1,1,1], index=[1,2,3])
ser2 = pd.Series([1,1,1], index=[3,4,5])
ser1 + ser2
```
would be, and _nobody_ got it right (😱 )